src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -1,19 +1,45 @@
 (ns puppetlabs.puppetdb.http.command
   (:require [puppetlabs.puppetdb.command.constants :refer [command-names]]
             [puppetlabs.trapperkeeper.core :refer [defservice]]
+            [clojure.string :as str]
             [clojure.tools.logging :as log]
             [puppetlabs.puppetdb.command :as command]
             [puppetlabs.puppetdb.http :as http]
+            [puppetlabs.puppetdb.cheshire :as json]
             [net.cgrand.moustache :as moustache]
             [puppetlabs.puppetdb.middleware :as mid]
             [compojure.core :as compojure]))
 
+(def min-supported-commands
+  {"replace catalog" 6
+   "replace facts" 4
+   "store report" 5
+   "deactivate node" 3})
```

Comment:
even if so, the defmulti contains versions that are not supported -- namely the ones being detected here. Is there a reason you are against using the minimum? Using a list of supported versions will also require a different error if the version is too high, since it wouldn't be retired in that case. As implemented here that's just handled the way it always has been.